### PR TITLE
Contributor docs

### PR DESCRIPTION
**Pull request by theref**
**Type of PR:**

- [x] Documentation

**Required reviews:**

- [ ] 1
- [x] 2

**What this does:**
> Extends `CONTRIBUTING.md` to make it suitable for onboarding new developers
> Also acts as a reference for existing devs

**Issues fixed/closed:**
> - Fixes https://github.com/nucypher/sprints/issues/27

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -19,15 +31,131 @@ pnpm lint
...
+
+2. Flexibility:
+
+    The system can accommodate various types of authentication providers, making it extensible and adaptable to different authentication needs.
+
+3. User Context Management:
+
+    They help manage user context, specifically the `:userAddress` parameter. This includes handling authentication tokens and ensuring that users are authenticated only once for multiple actions.
+
+Currently, [SIWE](https://docs.login.xyz/) (Sign In With Ethereum, [EIP-4361](https://eips.ethereum.org/EIPS/eip-4361)) is the only authentication method implemented.
```
Comment (manumonti) [inline CONTRIBUTING.md:106]:
```suggestion
Currently, [SIWE](https://docs.login.xyz/) (Sign-In With Ethereum, [EIP-4361](https://eips.ethereum.org/EIPS/eip-4361)) is the only authentication method implemented.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -19,15 +31,131 @@ pnpm lint
...
+EIP-712 has previously been supported but is now deprecated.
+
+[The below test](https://github.com/nucypher/taco-web/blob/b689493a37bec0b168f80f43347818095c3dd5ce/packages/taco/test/conditions/context.test.ts#L382C1-L429C6) demonstrates how a SIWE message can be reused for TACo authentication.
+This ensures that users don't have to sign multiple messages when logging into apps and decrypting TACo messages.
```
Comment (manumonti) [inline CONTRIBUTING.md:110]: I would put it this way, WDYT?

```suggestion
Reusing a SIWE authentication message allows the users to log in to the Dapp so that this authentication is also used for TACo decryption, i.e. users only have to sign authentication messages once.
```

**Review by manumonti — CHANGES_REQUESTED** [review]
I have some suggestions. I leave it up to you to considere whether they are right or not.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -19,15 +31,131 @@ pnpm lint
...
 ```
 
-## Publishing
+### Publishing
 
 TODO: Update after implementing automated publishing.
+
+# External Api
+This is the api that we expose to developers.
+It is defined in [`packages/taco/src/taco.ts`](https://github.com/nucypher/taco-web/blob/main/packages/taco/src/taco.ts)
```
[the remark this answers is not in the document]
Comment (theref) [reply CONTRIBUTING.md:49]: no, this will break the link

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -19,15 +31,131 @@ pnpm lint
...
+
+# Design and Architecture
+
+## Context Parameters
+When defining conditions for decryption, context parameters are specified to indicate what information is required.
+These parameters are placeholders that will be filled with actual data during the decryption process.
+
+The built in `:userAddress` parameter is defined in `packages/taco-auth/src/auth-provider.ts`.
```
[the remark this answers is not in the document]
Comment (theref) [reply CONTRIBUTING.md:57]: links like that don't work

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -19,15 +31,149 @@ pnpm lint
...
+npm dist-tag add @nucypher/taco@0.3.0 devnet
+```
+
+Ideally, this would all be automated using Github actions, workflows, releases, and tags.
+
+# External API
+This is the api that we expose to developers.
+It is defined in [`packages/taco/src/taco.ts`](https://github.com/nucypher/taco-web/blob/main/packages/taco/src/taco.ts)
+
+# Design and Architecture
```
Comment (piotr-roslaniec) [inline CONTRIBUTING.md:69]: Is [`CONTRIBUTING.md`](https://contributing.md/how-to-build-contributing-md/) the right place to discuss the design and architecture? Should we move this to Threshold Docs or `taco` package docs?

File: CONTRIBUTING.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (piotr-roslaniec) [reply CONTRIBUTING.md:75]: They [do](https://github.com/orgs/community/discussions/23148), please see [this example](https://github.com/nucypher/taco-web/tree/main/packages/taco#nucyphertaco-web)

**Comment by piotr-roslaniec** [comment]
We may also want to add a note about https://www.conventionalcommits.org/en/v1.0.0/

File: CONTRIBUTING.md (hunk 5 above)
Comment (theref) [reply CONTRIBUTING.md:69]: It's possibly the wrong heading - maybe Structure? it's supposed to be a section that helps new contributors get to grips with how the whole repo is laid out and organised

Comment (piotr-roslaniec) [reply CONTRIBUTING.md:73]: If it has to do with repo structure, we should describe how the `pnpm` workspace works, not API design

File: packages/taco/README.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # `@nucypher/taco`
 
-### [`nucypher/taco-web`](../../README.md)
+### [`nucypher/taco-web`](https://github.com/nucypher/taco-web/blob/main/README.md)
```
Comment (piotr-roslaniec) [inline packages/taco/README.md:3]: Why this change? Didn't the relative link work?
